install/helm/agones/templates/pdb.yaml

```diff
@@ -0,0 +1,27 @@
+# Copyright 2022 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{- range .Values.gameservers.namespaces }}
+---
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: agones-gameserver-safe-to-evict-false
+  namespace: {{ . }}
+spec:
+  maxUnavailable: 0%
+  selector:
+    matchLabels:
+      agones.dev/safe-to-evict: "false"
```

Comment:
Also open to bikeshedding here, as it's introducing a bit of an API contract. I called it `safe-to-evict` because you can essentially replace `safe-to-evict` with it (CA won't touch PDB covered pods), but we could also call it something more direct like `agones.dev/full-availability` or somesuch (avoiding double negatives with this approach is super hard).

Reply:
Quite like "safe-to-evict" -- but as above, if we feature flag all this, it becomes less of a burden to get this right out of the gate, and we can iterate as we go.